src/message.rs

```diff
@@ -3,6 +3,12 @@ use std::{fmt, mem, ptr};
 use super::{ffi, Error, MessageType, TypeSig, libc, to_c_str, c_str_to_slice, init_dbus};
 use std::os::unix::io::{RawFd, AsRawFd};
 
+#[derive(Debug)]
+pub enum ArrayError {
```

Comment:
ArrayError should derive Copy and Clone, and probably needs to be reexported as well?